README.md

```diff
@@ -55,7 +55,119 @@ With the default configurations defined in this repo's `.env.*` files, the porta
+
+#### Generating and publishing a did:web
+
+An example DID (Alice):
+
+```json
+{
+  "@context": [
+    "https://www.w3.org/ns/did/v1",
+    "https://w3id.org/security/suites/jws-2020/v1"
+  ],
+  "id": "did:web:jonmattgray.github.io:dids:test:alice",
+  "verificationMethod": [
+    {
+      "id": "did:web:jonmattgray.github.io:dids:test:alice#owner",
+      "type": "JsonWebKey2020",
+      "controller": "did:web:jonmattgray.github.io:dids:test:alice",
+      "publicKeyJwk": {
+        "kty": "OKP",
+        "crv": "Ed25519",
+        "x": "PUBLIC_KEY"
+      }
+    }
+  ],
+  "authentication": ["did:web:jonmattgray.github.io:dids:test:alice#owner"],
+  "assertionMethod": ["did:web:jonmattgray.github.io:dids:test:alice#owner"],
+  "service": [
+    {
+      "id": "did:web:jonmattgray.github.io:dids:test:alice#did-communication",
+      "type": "did-communication",
+      "priority": 0,
+      "recipientKeys": ["did:web:jonmattgray.github.io:dids:test:alice#owner"],
+      "routingKeys": [],
+      "accept": ["didcomm/aip1"],
+      "serviceEndpoint": "http://nice-agent-alice-veritable:5002"
+    }
+  ]
+}
+```
+
+`"serviceEndpoint": "http://nice-agent-alice-veritable:5002"` matches the hostname of Alice's cloudagent container. The endpoint is used when resolving the DID.
+
+The service endpoint for each persona is:
+
+- http://nice-agent-alice-veritable:5002
+- http://nice-agent-bob-veritable:5002
+- http://nice-agent-issuer-veritable:5002
+
+Use https://mkjwk.org/ with `key type: OKP` and `curve: Ed25519` to generate a key-pair for each DID.
+
+```json
+{
+  "kty": "OKP",
+  "d": "PRIVATE_KEY",
+  "crv": "Ed25519",
+  "x": "PUBLIC_KEY"
+}
+```
+
+Use the `x` value for `"x": "PUBLIC_KEY"` in the DID. Save the private `d` value to be used later.
+
+GitHub Pages can be used to host web DIDs for free:
+
+1.  Create a public repo called `YOUR_USERNAME.github.io`
+2.  In the repo, create your desired directory structure e.g. `dids/test/alice`
+3.  Add a `did.json` in the nested directory.
+4.  After pushing to GitHub, the DID can be viewed e.g. https://jonmattgray.github.io/dids/test/alice/did.json. Note the directory structure matches the did `did:web:jonmattgray.github.io:dids:test:alice`.
+
+The DID and PRIVATE_KEY of each persona is then set using env files at the root of this repo. For example:
+
+`.env.nice-agent-alice.local`
+
+```
+DID=did:web:jonmattgray.github.io:dids:test:alice
+PRIVATE_KEY=PRIVATE_KEY
+```
+
+`.env.nice-agent-bob.local`
+
+```
+DID=did:web:jonmattgray.github.io:dids:test:bob
+PRIVATE_KEY=PRIVATE_KEY
+```
+
+`.env.nice-agent-issuer.local`
+
+```
+DID=did:web:jonmattgray.github.io:dids:test:charlie
+PRIVATE_KEY=PRIVATE_KEY
+```
+
+#### Importing DIDs
+
+Finally, before forming connections, each DID needs to be imported to its cloudagent using the `dids/import` endpoint. Run the following script, which automatically uses DIDs and private keys from the `.local` env files to import all three personas:
```

Comment:
I know this is over-explaining but we could mention that one should have all the containers up before trying to run the script

Reply:
added `With containers up using 'npm run compose:up:test'...`